src/tapis-hooks/files/useMkdir.ts

```diff
@@ -0,0 +1,41 @@
+import { useEffect } from 'react';
+import { useMutation } from 'react-query';
+import { Files } from '@tapis/tapis-typescript';
+import { mkdir } from 'tapis-api/files';
+import { useTapisConfig } from 'tapis-hooks';
+import QueryKeys from './queryKeys';
+
+const useMkdir = (systemId: string, path: string) => {
+  const { basePath, accessToken } = useTapisConfig();
+  const jwt = accessToken?.access_token || '';
+
+  // The useMutation react-query hook is used to call operations that make server-side changes
+  // (Other hooks would be used for data retrieval)
+  //
+  // In this case, mkdir helper is called to perform the operation
+  const { mutate, isLoading, isError, isSuccess, data, error, reset } =
+    useMutation<Files.FileStringResponse, Error, Files.MkdirOperationRequest>(
```

Comment:
Replace Files.MkdirOperationRequest in the generic with a custom type

```
type MkdirParams = { systemId: string, path: string }
```

Refer to useLogin

https://github.com/tapis-project/tapis-ui/blob/main/src/tapis-hooks/authenticator/useLogin.ts